src/state/slices/marketDataSlice/selectors.ts

```diff
@@ -41,6 +43,26 @@ export const selectPriceHistoriesLoadingByAssetTimeframe = createSelector(
     !assetIds.every(assetId => Boolean(priceHistory[timeframe][assetId]))
 )
 
+const selectPriceHistoryQueryByAssetTimeframe = createSelector(
+  (_state: ReduxState, assetId: CAIP19, _timeframe: HistoryTimeframe) => assetId,
+  (_state: ReduxState, _assetId: CAIP19, timeframe: HistoryTimeframe) => timeframe,
+  (assetId, timeframe) =>
+    marketApi.endpoints.findPriceHistoryByCaip19.select({ assetId, timeframe })
```

Comment:
we can't do this pattern here - selectors can't be dispatching actions.

have a look at the docs for RTK query - `findPriceHistoryByCaip19` exports automatically generated hooks that we can consume to get a loading state.

Reply:
the `marketApi.endpoints.findPriceHistoryByCaip19.select` is a SelectorFactory, it doesn't dispatch any action.

I also prefer to use the hook to get the loading state, but the original issue explicitly ask for making it a selector (which may make it a cleaner pattern)